src/transform/globalize.js

```diff
@@ -29,30 +28,75 @@ function indent (code) {
   return lines.join('\n');
 }
 
-function relative (file) {
+function makePathRelative (file) {
   return path.relative(process.cwd(), file);
 }
 
+function defineDependencies (imports, dependencies) {
+  var code = '';
+  var keyVals = imports.map(function (imp, idx) {
+    return '"' + imp + '": ' + generateModuleName(dependencies[idx]);
+  });
+
+  keyVals.unshift('"exports": exports');
+  keyVals.unshift('"module": module');
+  code = '\n' + indent(keyVals.join(',\n')) + '\n';
+  return '{' + code + '}';
+}
+
+function defineReplacement (name, deps, func) {
+  var rval;
+  func = [func, deps, name].filter(function (cur) { return typeof cur === 'function'; })[0];
+  deps = [deps, name, []].filter(Array.isArray)[0];
+  rval = func.apply(null, deps.map(function (value) {
+    return defineDependencies[value];
+  }));
+  return rval && (exports = module.exports = rval);
+}
+
 module.exports = function () {
   return function (data, info) {
-    var sourcemap = data.match(regexSourcemap);
-    sourcemap = sourcemap && sourcemap[1];
+    var isAmd = data.match(regexAmd);
+    var vars = [];
 
-    if (sourcemap) {
-      data = data.replace(regexSourcemap, '');
-    }
+    // Strict mode can cause problems with dependencies that you don't have
+    // control over. Assume the worst.
+    data = data.replace(regexUseStrict, '');
 
+    // Replace all requires with references to dependency globals.
     info.imports.forEach(function (imp, index) {
       data = data.replace('require("' + imp + '")', generateModuleName(info.dependencies[index]));
     });
 
-    data = data.replace(regexUseStrict, '');
-    data = 'var module = { exports: {} };\nvar exports = module.exports;\n\n' + data;
+    // We assume CommonJS because that's what we're using to convert it.
+    vars.push('var module = {\n' + indent('exports: {}') + '\n};');
+    vars.push('var exports = module.exports;');
+
+    // We only need to generate the AMD -> CommonJS shim if it's used.
+    if (isAmd) {
+      vars.push('var defineDependencies = ' + defineDependencies(info.imports, info.dependencies) + ';');
+      vars.push('var define = ' + defineReplacement + ';');
+      vars.push('define.amd = true;');
```

Comment:
Maybe the name of this can be changed since it's not just `vars` now? Something like `injectedLines` / `shim`?